### ISSUE TITLE
Cannot run eval using model

### ISSUE THREAD

**Issue by mdtlab200s** [wvangansbeke/Unsupervised-Classification#27]
Hello,

First of all thank you for sharing your research to the public.   This is interesting work!

I am new to this and please forgive me if I am asking a simple question.  

I ran your scripts and it all worked as expected.  At the end of self-labeling step my result is this

... tail end of the results ..
Epoch 200/200
--
Adjusted learning rate to 0.00010
Train ...
Epoch: [199][ 0/50]	Loss 7.1724e-02 (7.1724e-02)
Epoch: [199][25/50]	Loss 5.5752e-02 (6.9237e-02)
Evaluate ...
{'ACC': 0.8724, 'ARI': 0.751001763292155, 'NMI': 0.7818230363066652, 'ACC Top-5': 0.9878, 'hungarian_match': [(0, 9), (1, 8), (2, 7), (3, 4), (4, 1), (5, 2), (6, 3), (7, 0), (8, 6), (9, 5)]}
Checkpoint ...
Evaluate model at the end
{'ACC': 0.8724, 'ARI': 0.751001763292155, 'NMI': 0.7818230363066652, 'ACC Top-5': 0.9878, 'hungarian_match': [(0, 9), (1, 8), (2, 7), (3, 4), (4, 1), (5, 2), (6, 3), (7, 0), (8, 6), (9, 5)]}

In my results directory (./results/cifar-10/selflabel/) i have 3 files , checkpoint.pth.tar,  confusion_matrix.png and model.pth.tar.

I tried running 'eval'  using this  `!python3 eval.py --config_exp configs/scan/scan_cifar10.yml  --visualize_prototypes --model ./results/cifar-10/selflabel/model.pth.tar` but got this error:

Load model weights ...
Traceback (most recent call last):
  File "eval.py", line 145, in <module>
    main() 
  File "eval.py", line 52, in main
    model.load_state_dict(state_dict['model'])
KeyError: 'model'

I tried using the same way from the self-label in the Model zoo google drive share.  What am i doing wrong here?  I want to see the 'visualize-prototypes' functionality.

**Comment by wvangansbeke** [comment]
Hi @mdtlab200s,

Make sure that you run the correct config file with the correct model. If you want to run `eval.py` after the self-labeling step, you should use the config file `config/selflabel/selflabel_cifar10.yml` instead of the one for scan.  You should indeed be able to visualize the prototypes then. Keep in mind that there is no guarantee that the prototypes will align perfectly with the ground truth. I normally visualize the prototypes after the scan-step but it shouldn't matter much. Btw, the models in the model zoo should give you the exact same prototypes for the validation set as we included in the paper. 

Hope this helps.

**Comment by mdtlab200s** [comment]
@wvangansbeke 

I ran these `!python3 eval.py --config_exp configs/scan/scan_cifar10.yml  --visualize_prototypes --model configs/selflabel/selflabel_cifar10.yml` but got this error.

Load model weights ...
Traceback (most recent call last):
  File "eval.py", line 145, in <module>
    main() 
  File "eval.py", line 46, in main
    state_dict = torch.load(args.model, map_location='cpu')
  File "/opt/tljh/user/lib/python3.7/site-packages/torch/serialization.py", line 593, in load
    return _legacy_load(opened_file, map_location, pickle_module, **pickle_load_args)
  File "/opt/tljh/user/lib/python3.7/site-packages/torch/serialization.py", line 763, in _legacy_load
    magic_number = pickle_module.load(f, **pickle_load_args)
_pickle.UnpicklingError: invalid load key, '#'.

**Comment by wvangansbeke** [comment]
Hi @mdtlab200s,

I actually mean this: `python3 eval.py --config_exp configs/selflabel/selflabel_cifar10.yml --model ./results/cifar-10/selflabel/model.pth.tar` --visualize_prototypes`. Your config file and model should match.

**Comment by mdtlab200s** [comment]
Hi @wvangansbeke ,

Thats what i did in my original post and got that error. I re-ran it again.

python3 eval.py --config_exp configs/scan/scan_cifar10.yml  --model results/cifar-10/selflabel/model.pth.tar --visualize_prototypes

and got this error:

Load model weights ...
Traceback (most recent call last):
  File "eval.py", line 145, in <module>
    main() 
  File "eval.py", line 52, in main
    model.load_state_dict(state_dict['model'])
KeyError: 'model'

**Comment by mdtlab200s** [comment]
Thanks, that seems to work. I dont get anymore error . I just need to figure out how to view those images in jupyter.  ALl i got are

/pytorch/aten/src/ATen/native/LegacyDefinitions.cpp:29: UserWarning: masked_select received a mask with dtype torch.uint8, this behavior is now deprecated,please use a mask with dtype torch.bool instead.
Figure(640x480)
Figure(640x480)
Figure(640x480)
Figure(640x480)
Figure(640x480)
Figure(640x480)
Figure(640x480)
Figure(640x480)
Figure(640x480)
Figure(640x480)

**Comment by wvangansbeke** [comment]
I never actually tried to run this code on Jupyter notebook. You probably need the magic command `%matplotlib inline` or someting similar. Just a guess though, it has been a very long time since I've used Jupyter notebook. I assume you will figure it out.

**Comment by mdtlab200s** [comment]
thank you again.  I will figure it out..

**Comment by shyam-98** [comment]
Hey @mdtlab200s, did you figure out how to solve the issue? I am also getting <Figure size 640x480 with 1 Axes> ... as output in jupyter notebook. I added the magic command but still it didn't work. So next, I tried to save the images by modifying code in the eval.py file (basically added `plt.savefig()` at the end of `visualize_indices` function but all I am getting are empty image files. Any help is greatly appreciated.
